sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -59,15 +73,20 @@ namespace Azure { namespace Core { namespace Http {
 
     /**
      * @brief Read portion of data into a buffer.
-     * @remark Throws if error/canceled.
+     *
+     * @remark Check if context is cancelled and then calls the read implementation.
      *
      * @param conntext #Context so that operation can be canceled.
      * @param buffer Pointer to a first byte of the byte buffer to read the data into.
      * @param count Size of the buffer to read the data into.
      *
      * @return Number of bytes read.
      */
-    virtual int64_t Read(Context const& context, uint8_t* buffer, int64_t count) = 0;
+    int64_t Read(Context const& context, uint8_t* buffer, int64_t count)
```

Comment:
Why not keep this virtual so someone can override it if they want to, but leave it as **non-pure** virtual?

Reply:
The main intention is to ensure to end-users that no matter what derived-bodyStream-class they use, the bodyStream will check for cancellation as the first instruction when calling `Read()`.
If we make this virtual, someone could remove the cancellation check and break this guarantee.

Reply:
It should not be virtual so we can enforce this behavior.